lib/galaxy/tools/loader_directory.py

```diff
@@ -156,11 +171,19 @@ def looks_like_a_tool_cwl(path):
         return False
 
     file_class = as_dict.get("class", None)
+    if classes is None or file_class not in classes:
+        return False
```

Comment:
I think instead of:

```
    if classes is None or file_class not in classes:
        return None
```

I probably meant 

```
    if classes is not None and file_class not in classes:
        return None
```

that way if no classes are chosen as filters, it would work for any class - still potentially a useful operation.